### PR TITLE
Add mapreduce for DArrays

### PR DESCRIPTION
**Pull request by samuela**

### CODE REVIEW

**Comment by ivarne** [comment]
Sorry this didn't get any response. Sometimes it isn't obvious who should review, and everybody just looks at it and think someone else will handle it. Bumping the issue/PR after a few days usually helps :smile:

I'm not much into the parallel functionality, but generally a new method should be accompanied with a few tests for both the main behaviour and a few edge cases.

**Comment by samuela** [comment]
Ok, added some tests with 0b86010.

File: test/parallel.jl [hunk 1]
```diff
@@ -34,6 +34,21 @@ d2 = map(x->1, d)
...
+    sig = methods(mapreduce, (Function, Function, DArray))[1].sig
+    @test sig[3] == DArray
+
+    # Test that it is functionally equivalent to the standard method
+    for _ = 1:25, f = [x -> 2x, x -> x^2, sin], opt = [+, *]
+        n = rand(2:50)
+        arr = rand(1:100, n)
+        darr = distribute(arr)
+
+        @test mapreduce(f, opt, arr) == mapreduce(f, opt, darr)
```
Comment (andreasnoack) [inline test/parallel.jl]: If you want to test with `==` here then you cannot use `sin` in the test. Either you'll have to use `@test_approx_eq` to test for almost equality or you'll have to use a function that doesn't promote to floating point.

Comment (samuela) [inline test/parallel.jl]: Oh, right. I forgot about that. For some reason these tests pass without an issue on my machine.

Comment (samuela) [inline test/parallel.jl]: Thanks for pointing that out @andreasnoack ! Should be actually fixed in 95f1fc9.

**Comment by andreasnoack** [comment]
Great. Thanks!

**Pull request merged**